magefile.go

```diff
@@ -114,6 +115,62 @@ func Build() error {
 	return nil
 }
 
+func BuildImages() error {
+	fmt.Println("Building images with ko...")
+
+	gitVersion := getVersion()
+	gitCommit := getCommit()
+	os.Setenv("BOM_LDFLAGS", generateLDFlags())
+
+	if os.Getenv("KO_DOCKER_REPO") == "" {
+		return errors.New("missing KO_DOCKER_REPO environment variable")
+	}
+
+	return sh.RunV("ko", "publish", "--base-import-paths", "--bare",
+		"--platform=all", "--tags", gitVersion, "--tags", gitCommit,
+		"sigs.k8s.io/bom/cmd/bom")
+}
+
+// BuildImagesLocal build images locally and not push
+func BuildImagesLocal() error {
+	fmt.Println("Building image with ko for local test...")
+	if err := EnsureKO(""); err != nil {
+		return err
+	}
+
+	os.Setenv("BOM_LDFLAGS", generateLDFlags())
+
+	return sh.RunV("ko", "publish", "--base-import-paths", "--bare",
+		"--local", "--platform=linux/amd64",
+		"sigs.k8s.io/bom/cmd/bom")
+}
+
+func BuildStaging() error {
+	if err := EnsureKO(""); err != nil {
+		return err
+	}
+
+	if err := Build(); err != nil {
+		return errors.Wrap(err, "building the binaries")
+	}
+
+	if err := BuildImages(); err != nil {
+		return errors.Wrap(err, "building the images")
+	}
+
+	if err := sh.RunV("cd", "output"); err != nil {
+		return errors.Wrap(err, "cd into output directory")
+	}
+
+	if err := sh.RunV("./bom-linux-amd64", "output", "generate", "-f", "bom-darwin-amd64",
+		"-f", "bom-darwin-arm64", "-f", "bom-linux-amd64", "-f", "bom-linux-arm64",
+		"-f", "bom-windows-amd64.exe", "-d", "../", "-o", "bom-sbom.sdpx"); err != nil {
+		return errors.Wrap(err, "generating the bom")
```

Comment:
This part is missing the images generated by ko before. But one thing to note is that to be able to add them we would need to alter the SBOM afterwards with the final registry.

The other option is that we wire SBOM generation into the promoter and attach the SBOMs with the signatures. I like the idea of having micro sboms in the images.

Reply:
sounds good, but this one is just for the binaries right? no images here

how do you want to proceed? we drop the bom generation from here and do at the promotion?

Reply:
This one will create an SBOM with the binaries, source code and dependencies. Lets keep it here and work on attaching micro SBOMs to the images when promoted. I think I like that approach better :)

Reply:
i will remove this part for now and we can discuss that later, I would like to get the job running to we see that working :)